client/albion_watcher.go

```diff
@@ -0,0 +1,76 @@
+package client
+
+import (
+	"log"
+	"time"
+
+	"github.com/google/gopacket/pcap"
+)
+
+type albionProcessWatcher struct {
+	pid       int
+	known     []int
+	listeners map[int][]*listener
+	quit      chan bool
+}
+
+func newAlbionProcessWatcher(pid int) *albionProcessWatcher {
+	return &albionProcessWatcher{
+		pid:  pid,
+		listeners: make(map[int][]*listener),
+		quit: make(chan bool),
+	}
+}
+
+func (apw *albionProcessWatcher) run() {
+	log.Printf("Watching Albion process with PID \"%d\"...", apw.pid)
+	r := newRouter()
+
+	for {
+		select {
+		case <-apw.quit:
```

Comment:
I feel like by removing PollChannel() and indenting all the code into a default this makes the code flow harder to read. The intent is just to check if a signal has been omitted into apw.quit and quit early if it has, but that intent is hidden in the boiler plate of the GO select statement.